File: src/my_project/easy_problems/from151To200/robot_return_to_origin.py
from typing import List 

class Solution:
    def judgeCircle(self, moves: str) -> bool:


        coordinate_position = {'L':-1, 'R':1, 'U':1,'D':-1}

        final_position: list[int] = [0,0]

        for move in moves: 
            if move == 'L' or move == 'R':
                final_position[0] += coordinate_position[move]
            else: 
                final_position[1] += coordinate_position[move]
                
        
        return final_position[0] == 0 and final_position[1] == 0

solution = Solution()

print(solution.judgeCircle(moves = "UD"))

print(solution.judgeCircle(moves = "LL"))


print(solution.judgeCircle(moves = "RRDD"))

print(solution.judgeCircle(moves = "LDRRLRUULR"))

File: my_project_typescript/easy_problems_typescript/from51To100/robotReturnToOrigin.ts
function judgeCircle(moves: string): boolean {

    let coordinatePosition: any = {'L':-1, 'R':1,'U':1,'D':-1}

    let finalPosition: Array<number> = [0,0]

    for (let move of moves){
        if (move == 'L' || move == 'R'){
            finalPosition[0] += coordinatePosition[move]
        }
        else {
            finalPosition[1] += coordinatePosition[move]

        }
    }

    return finalPosition[0] === 0 && finalPosition[1] == 0

};

console.log(judgeCircle("UD"))

console.log(judgeCircle("LL"))

console.log(judgeCircle("RRDD"))

console.log(judgeCircle( "LDRRLRUULR"))
